internal/cache/controller_obj_cache.go

```diff
@@ -355,7 +355,12 @@ func (c *AviObjCache) DeleteUnmarked(childCollection []string) {
 	for _, objkey := range c.SSLKeyCache.AviGetAllKeys() {
 		intf, _ := c.SSLKeyCache.AviCacheGet(objkey)
 		if obj, ok := intf.(*AviSSLCache); ok {
-			if obj.HasReference == false && obj.Name != lib.GetIstioWorkloadCertificateName() {
+			if obj.HasReference == false {
+				// if deleteConfig is false and istio is enabled, do not delete istio sslkeycert
+				if !lib.GetDeleteConfigMap() && lib.IsIstioEnabled() &&
```

Comment:
Should we optimize this condition as `obj.Name == lib.GetIstioWorkloadCertificateName() && !lib.GetDeleteConfigMap() && lib.IsIstioEnabled()` 
Reason: With current condition, For most of the objects, first two conditions will be true (when deleteConfig is false and istio is disabled) and then last condition will be false. 
So we can avoid processing these two conditions for remaining objects

Reply:
Similar changes at other places.

Reply:
changed it to have the order of rarity of encounter, name > istioEnabled true > deleteConfig false